File: packages/python/src/trend_buddy_shared/contracts/views/backtest.py
from dataclasses import dataclass, field
from datetime import datetime
from decimal import Decimal

from trend_buddy_shared.contracts.domain.execution import Fill
from trend_buddy_shared.contracts.views.analytics import PerformanceSummary


@dataclass
class IndicatorPoint:
    timestamp: datetime
    value: Decimal


@dataclass
class IndicatorSeries:
    indicator: str
    values: list[IndicatorPoint] = field(default_factory=list)


@dataclass
class EquityPoint:
    timestamp: datetime
    equity: Decimal


@dataclass
class BacktestResult:
    run_id: str
    strategy_id: str
    summary: PerformanceSummary
    fills: list[Fill] = field(default_factory=list)
    equity_curve: list[EquityPoint] = field(default_factory=list)

File: packages/go/contracts/views/backtest.go
package views

import (
	"time"

	"trend-buddy/packages/go/contracts/domain"
)

type IndicatorPoint struct {
	Timestamp time.Time      `json:"timestamp"`
	Value     domain.Decimal `json:"value"`
}

type IndicatorSeries struct {
	Indicator string           `json:"indicator"`
	Values    []IndicatorPoint `json:"values,omitempty"`
}

type EquityPoint struct {
	Timestamp time.Time      `json:"timestamp"`
	Equity    domain.Decimal `json:"equity"`
}

type BacktestResult struct {
	RunID       string             `json:"run_id"`
	StrategyID  domain.StrategyID  `json:"strategy_id"`
	Summary     PerformanceSummary `json:"summary"`
	Fills       []domain.Fill      `json:"fills,omitempty"`
	EquityCurve []EquityPoint      `json:"equity_curve,omitempty"`
}
